machine/v1beta1/types_vsphereprovider.go

```diff
@@ -193,13 +193,34 @@ type VSphereDisk struct {
 	// +required
 	Name string `json:"name"`
 	// sizeGiB is the size of the disk in GiB.
-	// The maximum supported size is 57742 GiB.
+	// The maximum supported size is 16384 GiB.
 	// +kubebuilder:validation:Minimum=1
-	// +kubebuilder:validation:Maximum=57742
+	// +kubebuilder:validation:Maximum=16384
```

Comment:
Why is this being decreased? Decreasing a maximum value is considered a breaking change as there may be instances of this resource that already specify values greater than 16384.

If you are looking to do ratcheting validation on this field, it is important to understand:
- How many instances of this resource that we keep track of have a value larger than 16384?
- Testing should be added to ensure the ratcheting validation works as expected
- The controller must continue to have logic that supports the original maximum to ensure you do not break existing users.

Reply:
This was accidentally put in here.  It was moved to a separate PR.